include/forcing/CsvPerFeatureForcingProvider.hpp

```diff
@@ -195,16 +195,16 @@ class CsvPerFeatureForcingProvider : public forcing::ForcingProvider
 
     // This map may be applicable to well-known netCDF formats as well?
     map<std::string, std::tuple<std::string, std::string>> well_known_fields = {
-        {"precip_rate", { CSDMS_STD_NAME_RAIN_RATE, "mm s^-1" } }, 
-        {"APCP_surface", { CSDMS_STD_NAME_LIQUID_EQ_PRECIP_RATE, "kg m^-2" } }, // Especially this one, is it correct? 
+        {"precip_rate", { CSDMS_STD_NAME_LIQUID_EQ_PRECIP_RATE, "mm s^-1" } }, 
+        {"APCP_surface", { CSDMS_STD_NAME_RAIN_RATE, "kg m^-2" } }, // Especially this one, is it correct? 
```

Comment:
I strongly recommend changing this macro in aorcforcing.hpp to be reflective of the variable it is mapped to:
```c++
#define CSDMS_STD_NAME_RAIN_RATE "atmosphere_water__rainfall_volume_flux"
```
Some quick dimensional analysis:
- volume flux implies dimensions of `mass^3 * time^-1 *  length^2`
- mass flux implies dimensions of `mass * time^-1`
- RAIN_RATE/precip_rate implies dimensions of `L * time^-1`

So, ACPC_surface, with units `kg * m^-2` is not even a flux.  To be a flux, there needs to be a time dimension.
I think RAIN_RATE and precip_rate both can map to CSDMS_STD_NAME_LIQUID_EQ_PRECIP_RATE (if its dimensions are `L * time^-1`.  ACPC_surface should probably map to a volume flux, but it cannot, without some unit of time, technically do so.  Now we do know that these are per hour, so perhaps we can put that assumption into this, but then I would correct the units of `APCP_surface` to be `kg * hr^-1 * m^-2` to make it a real volume flux.

Reply:
Just tell me what CSDMS name to make it... but it seems some model configs and/or realization configs will have to be adjusted to match.

Reply:
(we may make that a follow-on issue)

Reply:
```c++
#define CSDMS_STD_NAME_VOLUME_FLUX "atmosphere_water__rainfall_volume_flux"
#define CSDMS_STD_NAME_MASS_FLUX "atmosphere_water__precipitation_mass_flux"
```
and then, assuming we are ok with adjusting the units here to make it a flux???
```c++
"APCP_surface", { CSDMS_STD_NAME_VOLUME_FLUX, "kg h^-1 m^-2" } }
```
Can probably get away with just mapping to the flux with the same units for the time being and opening an issue to resolve later.  
```c++
"APCP_surface", { CSDMS_STD_NAME_VOLUME_FLUX, "kg m^-2" } }
```
Long story short, rain_rate should not be associated with APCP_surface in such a direct fashion.

Reply:
Oh. Didn't understand from the above that you wanted the macro _name_ changed. In either case, not really opposed, but those macro names are used elsewhere, scattered in the legacy `Forcing` class, in `Simple_Lumped_Model_Realization` and in the ET calculations in `Bmi_Module_Formulation`... so while it's a straightforward find and replace, it will bring in a lot more files. Is this in scope? Or should it be also a separate issue?

Reply:
So I S&R'd the `RAIN_RATE` macro and added the `MASS_FLUX` one (which isn't presently used anywhere)... I think that fulfills the central objection.

As for the units, from the CsvReader's point of view, it's just capturing knowledge about what the units in the file are based on a common column name... so whatever units are there should be whatever the units in the column are... so the real question is "what are the units in the [CSV] column?" (Same question should apply to the CSDMS name!) If they don't match the way they're being used, that's a separate issue... what's here is supposed to be what the CSV file represents.

I have a "TODO" in there to parse units from the headers... and really even the CSDMS names are a convenience... both of those things should arguably come out in the future or at least be overridable (but expanding the `forcing_config` struct would have been a heavier lift). Point being, if the units should be changed, they should be changed to whatever the source format is believed to be, and will probably be drawn more directly from it in the future anyway.